### ISSUE TITLE
Add relative units for font size option

### ISSUE THREAD

**Issue by iamtakashi** [WordPress/gutenberg#23323]
The custom font size option is really useful, but the lack of relative units has been a problem to make patterns work well in both large and small screens.

It'd be great if we could have some the relative units, just like the cover block height control.

<img width="129" alt="Screenshot 2020-06-19 at 18 39 24" src="https://user-images.githubusercontent.com/908665/85164454-4a35da00-b25c-11ea-82eb-0d63fa7f5cf5.png"> 

With the `vw` unit, this sort of type treatment would be possible, I think.

<img width="1187" alt="Screenshot 2020-06-19 at 18 41 34" src="https://user-images.githubusercontent.com/908665/85164714-a567cc80-b25c-11ea-8fc0-235c3e4c6ed8.png">

I've seen this topic has come up before, and I'll apologise if there is a better thread to discuss this. If there is please point me at there.

**Comment by carolinan** [comment]
Related: 
https://make.wordpress.org/themes/2020/05/20/block-based-theme-meeting-notes-20-may-2020/

https://github.com/WordPress/gutenberg/pull/22505

**Comment by aristath** [comment]
Expanding on this one, it would be useful to just have a freeform option.
Values don't necessarily need to be `10px`, `2em`, `3vw` etc... Some cases require more elaborate values.
Maybe I want to use `calc(14px + 0.5vw)` and have responsive typography. Or maybe I just want to use [`clamp()`](https://developer.mozilla.org/en-US/docs/Web/CSS/clamp). Or even just use a css-variable and type `var(--base-font-size)`.

Forcing users to pick a numeric value and a unit will cover the majority of cases, but it will still limit things and prevent valid use-cases that make sense.

**Comment by jorgefilipecosta** [comment]
I agree with what is being said here. Ideally, the font size picker component would be expanded to support string values.
When it can parse things like "1em" it would render relevant pickers with the correct units. If it can not parse because it is a complex value like calc(14px + 0.5vw) it would show a simple input field with the string.

The same would happen for all controls where we accept value/units or more complex CSS.

**Comment by mtias** [comment]
This is supported now.

**Comment by iamtakashi** [comment]
> This is supported now.

@mtias I can see `px`, `em`, and `rem` units available now, but `vw` and `vh` aren't yet available. Am I missing something? 

<img width="282" alt="Screenshot 2022-05-26 at 18 03 47" src="https://user-images.githubusercontent.com/908665/170539498-d9c3bab1-2ccf-43d1-94e9-a2b07721672d.png">

It depends on #24480, but if that will take more time, I think it's worth considering adding `vw` and `vh` as well. What do you think?

**Comment by mtias** [comment]
Ah, sorry, I misread :)

**Comment by iamtakashi** [comment]
That's ok. Thanks for reopening!

**Comment by flabernardez** [comment]
This option will be added soon? 

I really need vw vh and calc or var options to fonts size 😄

**Comment by carolinan** [comment]
@iamtakashi @flabernardez @beafialho @WordPress/block-themers @WordPress/outreach 
Would having vw and vh units in the custom font size picker* be useful even if it is not combined with clamp()?

(*The control in the Typography panel where you select a completely custom font size instead of using a preset)

**Comment by iamtakashi** [comment]
> Would having vw and vh units in the custom font size picker* be useful even if it is not combined with clamp()?

It can be useful. Here's a recent example of using the `vw` unit for this theme: https://surrealist.mystagingwebsite.com/. I see this kind of design treatment, a single word spanning the screen, more often in the wild lately. When the editor recognises the unit, the user can adjust the size easily depending on their content. 

However, while working on the theme, I realised that the editor already recognises the `vw` unit if it's first added in the code editor. I was pleasantly surprised!

`<!-- wp:site-title {"textAlign":"center","isLink":false,"style":{"typography":{"fontSize":"20vw"}}} /-->`

![CleanShot 2024-04-11 at 10 36 43@2x](https://github.com/WordPress/gutenberg/assets/908665/d46d54b0-89bb-4cc7-9c27-834f5121b92b)

**Comment by youknowriad** [comment]
Closing as this seems solved already.
